app/src/main/java/de/westnordost/streetcomplete/data/user/oauth/OAuthAuthorization.kt

```diff
@@ -39,12 +44,12 @@ import kotlin.io.encoding.ExperimentalEncodingApi
  *              several at once (using the same redirect URI)
  */
 @Serializable class OAuthAuthorization(
```

Comment:
Right, I should have kept in mind when I wrote this class that the HttpClient should be injected and hence the structure of this class cannot be kept that way.

I think it would be cleaner to make OAuthAuthorization a pure data class (maybe rename to OAuthAuthorizationParams) and move all logic into class that you named OAuthService.

Reply:
So, I've done a little bit of refactoring of this class. I've kept a few methods in the `OAuthAuthorizationParams` because they are highly coupled to the fields of that class, and nothing else, so it made sense to me to keep them in that class. Happy to do more moving around if you disagree though.